cirq-google/cirq_google/engine/engine_processor.py

```diff
@@ -105,19 +111,23 @@ def get_sampler(
         processor = self._inner_processor()
         # If a run_name or config_alias is not provided, initialize them
         # to the Processor's default values.
-        if not run_name and not device_config_name:
+        if not run_name and not device_config_name and not snapshot_id:
```

Comment:
curious why before we use `if (bool(run_name) or bool(snapshot_id)) ^ bool(device_config_name):`?